src/Altinn.App.Core/Features/Options/CommonOptionProviderServiceCollectionExtensions.cs

```diff
@@ -61,4 +66,53 @@ public static IServiceCollection AddAltinn2CodeList(
         ));
         return serviceCollection;
     }
+
+    /// <summary>
+    /// Extention method for IServiceCollection to add the AddAltinn3CodeList() method
+    /// <code>
+    /// services.AddAltinn3CodeList(
+    ///     optionId: "someNewCodeList-1",
+    ///     org: "ttd",
+    ///     codeListId: "someNewCodeList",
+    ///     version: "1"
+    /// );
+    /// </code>
+    /// </summary>
+    /// <param name="serviceCollection">The service collection to add altinn 3 codelists to</param>
+    /// <param name="optionId">
+    ///    The id/name that is used in the <c>optionsId</c> parameter in the SelectionComponents (Checkboxes, RadioButtons, Dropdown ...)
+    /// </param>
+    /// <param name="org">The organization that has this code list</param>
+    /// <param name="codeListId">Id of the code list in the code list repository.</param>
+    /// <param name="version">Version of the code list in the code list repository. Defaults to latest if not provided.</param>
+    /// <returns></returns>
+    public static IServiceCollection AddAltinn3CodeList(
+        this IServiceCollection serviceCollection,
+        string optionId,
+        string org,
+        string codeListId,
+        string? version = null
+    )
+    {
+        serviceCollection.AddHttpClient(
+            "Altinn3LibraryClient",
+            (serviceProvider, client) =>
+            {
+                var platformSettings = serviceProvider.GetRequiredService<IOptions<PlatformSettings>>().Value;
+                client.BaseAddress = new Uri(platformSettings.Altinn3LibraryApiEndpoint);
+                client.Timeout = TimeSpan.FromSeconds(30);
+            }
```

Comment:
Could we just skip the named client magic and configure the httpClient directly in the appOptionsProvider?